src/cpp/server/DBActiveSessionStorage.cpp

```diff
@@ -0,0 +1,436 @@
+                        a += ", '" + b.second + "'";
+                     }
+                     return a;
+                });
+      return values;
+
+   }
+
+   std::string getUpdateString(std::map<std::string, std::string> sourceMap)
+   {
+      std::string setValuesString = std::accumulate(++sourceMap.begin(), sourceMap.end(),
+                  sourceMap.begin()->first + " = '" + sourceMap.begin()->second + "'", [](std::string a, std::pair<std::string, std::string> iter){
+                     return a + ", " + iter.first + " = '" + iter.second + "'";
+                  });
+      return setValuesString;
+   }
+
+   std::string getColumnNameList(std::set<std::string> colNames)
+   {
+      std::string cols = std::accumulate(++colNames.begin(), colNames.end(), 
+                  *(colNames.begin()), [](std::string a, std::string b) {
+                  return a + ", " + b;
+                });
+      return cols;
+   }
+
+   void populateMapWithRow(database::RowsetIterator iter, std::map<std::string, std::string> *targetMap)
+   {
+      for(size_t i=0; i < iter->size(); i++)
+      {
+         std::string key = iter->get_properties(i).get_name();
+         if(key != kUserId)
+         {
+            targetMap->insert(
+               std::pair<std::string, std::string>{
+                  key,
+                  iter->get<std::string>(key, "")
+               }
+            );
+         }
+         else
+         {
+            targetMap->insert(
+               std::pair<std::string, std::string>{
+                  key,
+                  std::to_string(iter->get<int>(key))
+               }
+            );
+         }
+      }
+   }
+
+} // anonymous namespace
+
+   Error getConn(boost::shared_ptr<database::IConnection>* connection) {
+      bool success = server_core::database::getConnection(boost::posix_time::milliseconds(500), connection);
+
+      if(!success)
+      {
+         return Error{"FailedToAcquireConnection", errc::ConnectionFailed, "Failed to acquire a connection in 500 milliseconds.", ERROR_LOCATION};
+      }
+
+      return Success();
+   }
+
+   class ConnectionNotEstablished
+   {
+   };
+
+   DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId)
+      : sessionId_(sessionId)
+   {
+      
+   }
+
+   DBActiveSessionStorage::DBActiveSessionStorage(const std::string& sessionId, boost::shared_ptr<core::database::IConnection> overrideConnection)
+      : sessionId_(sessionId), overrideConnection_(overrideConnection)
+   {
+
+   }
+
+   Error DBActiveSessionStorage::readProperty(const std::string& name, std::string* pValue)
+   {
+      *pValue = "";
+      boost::shared_ptr<database::IConnection> connection;
+      Error error = Success();
+      if(overrideConnection_ == nullptr)
+      {
+         error = getConn(&connection);
+      }
+      else
+      {
+         connection = overrideConnection_;
+      }
+      
+      if(error)
+      {
+         return error;
+      }
+
```

Comment:
Nit: reduce line count a bit
```suggestion
      if(overrideConnection_ == nullptr)
         error = getConn(&connection);
      else
         connection = overrideConnection_;
      
      if(error)
         return error;
```